added new page: Data science (#1038)
arangodb/docs: 3.10/data-science.md

---
layout: default
description: Scalable Graph Analytics and GraphML
---
Data Science
==================

> 

Reviewer: A [lead paragraph](https://github.com/arangodb/docs#adding-a-lead-paragraph) would be nice, maybe using the first sentence of the content.
See e.g. https://github.com/arangodb/docs/blob/main/3.10/use-cases.md?plain=1#L3-L12 (ignore the `page-toc` part).

ArangoDB's Graph Analytics and GraphML capabilities provide various solutions
in data science and data analytics. Multiple data science personas within the
engineering space can make use of ArangoDB's set of tools and technologies that
enable analytics and machine learning on graph data. 

ArangoDB, as the foundation for GraphML, comes with the following key features:

> - **Scalable**: Designed to support true scalability with high performance for
>  Enterprise use cases
> - **Simple Ingestion**: Easy integration in existing data infrastructure with
>  connectors to all leading data processing and data ecosystems

Reviewer: Indentation should be the same for all lines
```suggestion
- **Scalable**: Designed to support true scalability with high performance for
  Enterprise use cases
- **Simple Ingestion**: Easy integration in existing data infrastructure with
  connectors to all leading data processing and data ecosystems
```

![Graph Analytics](images/graph-analytics.png)

### GraphML

When applying machine learning on a graph, you can predict connections, get 
better product recommendations, and also classify vertices, edges, and graphs.

GraphML can answer questions like 
- _**Is there a connection between person X and person Y?**_
- _**Will a customer churn?**_ 
- _**Is this particular transaction Anomalous?**_

![Graph ML](images/graph-ml.png)

## Use Cases

This section contains an overview of different use cases where Graph Analytics
and GraphML can be applied.

### Graph Analytics

Graph Analytics consists of an analysis applied on graph-based data and has
applicability in various fields such as marketing, fraud, supply chain,
product recommendations, drug development, law enforcement, and cybersecurity.

Graph Analytics uses an unsupervised
learning method based on algorithms that are performing analytical processing
directly on graphs stored in ArangoDB. This 
[Distributed Iterative Graph Processing (Pregel)](graphs-pregel.html)
is intended to help you gain analytical insights on
your data, without having to use external processing systems.

ArangoDB includes the following graph algorithms:

> - [Page Rank](graphs-pregel.html#page-rank): used for ranking documents in a graph
> Search/Traversal
> - [Single-Source Shortest Path](graphs-pregel.html#single-source-shortest-path): calculates
>  the shortest path length between the source and all other vertices.
>  For example, _How to get from a to b_
> - [Hyperlink-Induced Topic Search (HITS)](graphs-pregel.html#hyperlink-induced-topic-search-hits): 
>  a link analysis algorithm that rates Web pages
> - [Vertex Centrality](graphs-pregel.html#vertex-centrality): identifies the most important
>  nodes in a graph. For example, _Who are the influencers in a social network?_
> - [Community Detection](graphs-pregel.html#community-detection): identifies distinct subgroups
>  within a community structure

Reviewer: ```suggestion
- [Page Rank](graphs-pregel.html#page-rank): used for ranking documents in a graph
  search/traversal
- [Single-Source Shortest Path](graphs-pregel.html#single-source-shortest-path): calculates
  the shortest path length between the source and all other vertices.
  For example, _How to get from a to b_
- [Hyperlink-Induced Topic Search (HITS)](graphs-pregel.html#hyperlink-induced-topic-search-hits): 
  a link analysis algorithm that rates Web pages
- [Vertex Centrality](graphs-pregel.html#vertex-centrality): identifies the most important
  nodes in a graph. For example, _Who are the influencers in a social network?_
- [Community Detection](graphs-pregel.html#community-detection): identifies distinct subgroups
  within a community structure
```

- Predict relationships among customers, accounts

ArangoDB is using well-known GraphML frameworks like
[Deep Graph Library](https://www.dgl.ai){:target="_blank"}
and [PyTorch Geometric](https://pytorch-geometric.readthedocs.io/en/latest/){:target="_blank"} 
and connects to these external machine learning libraries. When coupled to
ArangoDB, you are essentially integrating them with your graph dataset.

## Example: ArangoFlix

ArangoFlix is a complete movie recommendation application that predicts missing
links between a user and the movies they have not watched yet.

This [interactive tutorial](https://colab.research.google.com/github/arangodb/interactive_tutorials/blob/master/notebooks/Integrate_ArangoDB_with_PyG.ipynb){:target="_blank"} 
demonstrates how to integrate ArangoDB with PyTorch Geometric to
build recommendation systems using Graph Neural Networks (GNNs).

The full ArangoFlix demo website is accessible from ArangoDB Oasis,
the managed cloud for ArangoDB. You can open the demo website that connects to
your running database from the **Examples** tab of your deployment.

{% hint 'tip' %}
You can try out ArangoDB Oasis free of charge for 14 days.

> Sign up [here](https://cloud.arangodb.com/){:target="_blank"}. 

Reviewer: "here" links should be avoided at all costs. It's better to describe what the link points to. Maybe change it to this?
```suggestion
Sign up at [cloud.arangodb.com](https://cloud.arangodb.com/){:target="_blank"}. 
```

---
layout: default
description: >- 

>    ArangoDB applies analytics and machine learning to graph data at scale

Reviewer: YAML indentation should use 2 spaces.
This sounds as if ArangoDB would autonomously process graphs without user interaction. How about this?
```suggestion
  ArangoDB lets you apply analytics and machine learning to graph data at scale
```
